### PR TITLE
Renyix/data import sdkv2

### PR DESCRIPTION
**Pull request by xiongrenyi**
# Description
Add SDK support for 'az ml data import --file dataimport.yaml' and 'az ml data show-materialization-status --name <asset_name>'

spec: https://github.com/Azure/azureml_run_specification/blob/master/specs/managed-data-asset/data-import-experience1.md

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_data_import/data_import.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from azure.ai.ml._utils._experimental import experimental
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY, PARAMS_OVERRIDE_KEY
+from azure.ai.ml.entities._assets import Data
+from azure.ai.ml.entities._inputs_outputs.external_data import Database, FileSystem
+from azure.ai.ml.entities._util import load_from_dict
+
+
+@experimental
+class DataImport(Data):
+    """Data import job to create the data asset.
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_data_import/data_import.py:20]: It's kind of strange to say this is Data import **job** here.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_data_import/data_import.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+from typing import Dict, Optional, Union
+from azure.ai.ml._schema import DataImportSchema
+from azure.ai.ml._utils._experimental import experimental
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY, PARAMS_OVERRIDE_KEY
+from azure.ai.ml.entities._assets import Data
+from azure.ai.ml.entities._inputs_outputs.external_data import Database, FileSystem
+from azure.ai.ml.entities._util import load_from_dict
+
+
+@experimental
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_data_import/data_import.py:18]: will this always print experiemental warning? as this is public exported?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py [hunk 3]
```diff
@@ -314,6 +320,67 @@ def create_or_update(self, data: Data) -> Data:
...
+            experiment_name=experiment_name,
+            compute="serverless",
+            source=data_import.source,
+            outputs={
+                "sink": Output(
+                    type=data_import.type, path=data_import.path, name=data_import.name, version=data_import.version
+                )
+            },
+            is_deterministic=False,
+            component="azureml://registries/azureml-dev/components/" + component_name + "/versions/1",
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py:353]: we can leave this empty here,  there is an default value in implementation of import_data

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py [hunk 4]
```diff
@@ -314,6 +320,67 @@ def create_or_update(self, data: Data) -> Data:
...
+        )
+        import_pipeline = PipelineJob(
+            description=data_import.description or experiment_name,
+            display_name=experiment_name,
+            experiment_name=experiment_name,
+            properties=data_import.properties or {},
+            jobs={experiment_name: import_job},
+        )
+        import_pipeline.properties["azureml.materializationAssetName"] = data_import.name
+        return self._job_operation.create_or_update(job=import_pipeline, skip_validation=True)
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py:363]: could you share submitted working job ink for this feature?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py [hunk 5]
```diff
@@ -314,6 +320,67 @@ def create_or_update(self, data: Data) -> Data:
...
+            display_name=experiment_name,
+            experiment_name=experiment_name,
+            compute="serverless",
+            source=data_import.source,
+            outputs={
+                "sink": Output(
+                    type=data_import.type, path=data_import.path, name=data_import.name, version=data_import.version
+                )
+            },
+            is_deterministic=False,
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py:352]: @YingChen1996 is is_deterministic supported?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py [hunk 6]
```diff
@@ -314,6 +320,67 @@ def create_or_update(self, data: Data) -> Data:
...
+
+        :param name: name of asset being created by the materialization jobs.
+        :type name: str
+        :param list_view_type: View type for including/excluding (for example) archived jobs. Default: ACTIVE_ONLY.
+        :type list_view_type: Optional[ListViewType]
+        :return: An iterator like instance of Job objects.
+        :rtype: ~azure.core.paging.ItemPaged[Job]
+        """
+
+        return self._job_operation.list(job_type="Pipeline", asset_name=name, list_view_type=list_view_type)
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py:382]: Is `asset_name` a supported param?

File: sdk/ml/azure-ai-ml/tests/data_import/e2etests/test_data_import.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    def test_data_import(self, client: MLClient) -> None:
+        data_import = load_data_import(
+            source="./tests/test_configs/data_import/data_import_e2e.yaml",
+        )
+        pipeline_job: PipelineJob = client.data.import_data(data_import)
+
+        key = "data_import_" + data_import.name
+        assert key in pipeline_job.jobs
+        assert isinstance(data_import, DataImport)
+        assert data_import.name == pipeline_job.jobs[key].outputs["sink"].name
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/data_import/e2etests/test_data_import.py:25]: may extract `pipeline_job.jobs[key]` as a variable.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_load_functions.py [hunk 8]
```diff
@@ -452,6 +453,37 @@ def load_data(
     return load_common(Data, source, relative_origin, **kwargs)
 
 
+def load_data_import(
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_load_functions.py:456]: I don't see sdk experience in this spec: https://github.com/Azure/azureml_run_specification/blob/master/specs/managed-data-asset/data-import-experience1.md

Could you point me to it?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py [hunk 9]
```diff
@@ -347,6 +353,63 @@ def create_or_update(self, data: Data) -> Data:
...
+            experiment_name=experiment_name,
+            properties=data_import.properties or {},
+            settings=PipelineJobSettings(force_rerun=True),
+            jobs={experiment_name: import_job},
+        )
+        import_pipeline.properties["azureml.materializationAssetName"] = data_import.name
+        return self._job_operation.create_or_update(job=import_pipeline, skip_validation=True)
+
+    @experimental
+    def show_materialization_status(
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py:395]: show_materialization_status the name feels strange for a function return a list of jobs

Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py:395]: looks like this is defined by the spec: https://github.com/Azure/azureml_run_specification/blob/master/specs/python-sdk/import-data-python.md

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_load_functions.py (hunk 8 above)
Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_load_functions.py:456]: https://github.com/Azure/azureml_run_specification/blob/master/specs/python-sdk/import-data-python.md

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py (hunk 9 above)
Comment (luigiw) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py:395]: I'm glad I'm not the only who feels this name is too long. :)

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py [hunk 10]
```diff
@@ -347,6 +353,63 @@ def create_or_update(self, data: Data) -> Data:
                     )
             raise ex
 
+    @experimental
+    def import_data(self, data_import: DataImport) -> Job:
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_data_operations.py:357]: The SDK syntax for this function is below, which feels redundant. We should consider to rename this function to be `import()`
```python
mlclient.data.import_data()
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_data_import/data_import.py [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+from pathlib import Path
+from typing import Dict, Optional, Union
+from azure.ai.ml._schema import DataImportSchema
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY, PARAMS_OVERRIDE_KEY
+from azure.ai.ml.entities._assets import Data
+from azure.ai.ml.entities._inputs_outputs.external_data import Database, FileSystem
+from azure.ai.ml.entities._util import load_from_dict
+
+
+class DataImport(Data):
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_data_import/data_import.py:17]: Please add an @experimental decorator to this class too.

**Review by luigiw — APPROVED** [review]

**Pull request merged**